apiserver/facades/client/action/action.go

```diff
@@ -119,20 +135,26 @@ func (a *ActionAPI) Actions(ctx context.Context, arg params.Entities) (params.Ac
 
 	response := params.ActionResults{Results: make([]params.ActionResult, len(arg.Entities))}
 	for i, entity := range arg.Entities {
-		currentResult := &response.Results[i]
-		tag, err := names.ParseTag(entity.Tag)
+		actionTag, err := names.ParseActionTag(entity.Tag)
 		if err != nil {
-			currentResult.Error = apiservererrors.ServerError(apiservererrors.ErrBadId)
+			response.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrBadId)
 			continue
 		}
-		_, ok := tag.(names.ActionTag)
-		if !ok {
-			currentResult.Error = apiservererrors.ServerError(apiservererrors.ErrBadId)
+
+		action, err := a.operationService.GetAction(ctx, coreaction.UUID(actionTag.Id()))
```

Comment:
question: there are two types of actionTags, ID and UUID. Which one is current per the cli? ID won't map to a UUID.

Reply:
The CLI validates with the names package: 
https://github.com/juju/juju/blob/a37b4500391a5dec4c60ba40fcee0bbf4f5b1986/cmd/juju/action/showtask.go#L113-L114
Which accepts both UUIDs and Numbers:
https://github.com/juju/names/blob/b50ca77a4137d8a23ec895d76d051d88248ecdee/action.go#L58-L64

Although I do see that the CLI examples only use (sequence?) numbers.

This needs clarification since we don't know what the tag will contain so therefore we cannot know if we need to first fetch a UUID with a number tag or just use the UUID we get. 

I can propose a patch deprecating the v1 action tag (UUID) on the names package.

In any case I think we could leave this for a follow up patch.